Tracker/DataStore/CategoryStore.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+
+final class CategoryStore: DataStoreDelegate {
+    
+    weak var delegate: DataStoreDelegate?
+    var dataStore: DataStore
+    var count: Int = 0
+    
+    init(dataStore: DataStore) {
+        self.dataStore = dataStore
+        dataStore.categoriesDelegate = self
+    }
+    
+    func getCategories() -> [Category] {
+        let categories = dataStore.categories
+        count = categories.count
+        return categories
+    }
+    
+    func getCategory(_ id: Int32) -> Category? {
+        do {
+            return try dataStore.getCategory(id)
+        } catch {
+            return nil
+        }
```

Comment:
Здесь можно пользоваться опциональным try: 
```suggestion
        return try? dataStore.getCategory(id)
```

Так как тебе не нужно обрабатывать ошибку, в случае таковой просто вернется nil. Можно поправить здесь и далее по проекту